lib/src/clojure_lsp/feature/diagnostics.clj

```diff
@@ -143,9 +144,15 @@
         (when (not= :unknown (shared/uri->file-type uri))
           (sync-lint-file! uri db))))))
 
+(defn pmap-light
+  "Call claypoole pmap with less threads than pmap to avoid topping cpu."
+  [f coll]
+  (let [threadpool-size (int (Math/ceil (/ (.. Runtime getRuntime availableProcessors) 3)))]
+    (cp/upmap threadpool-size f coll)))
```

Comment:
looks better having this function here indeed.
```suggestion
(defn ^:private pmap-light
  "Call claypoole pmap with less threads than pmap to avoid topping cpu."
  [f coll]
  (let [threadpool-size (int (Math/ceil (/ (.. Runtime getRuntime availableProcessors) 3)))]
    (cp/upmap threadpool-size f coll)))
```